test/test_pebble.py

```diff
@@ -985,11 +989,12 @@ def test_check_info(self):
         self.assertEqual(check.failures, 5)
         self.assertEqual(check.threshold, 3)
 
-        check = pebble.CheckInfo.from_dict({
+        d: pebble._CheckInfoDict = {
             'name': 'chk3',
             'status': 'up',
             'threshold': 3,
-        })
+        }  # type: ignore (no explicit level or failures)
```

Comment:
Ah yes, this is a bit annoying, isn't it? I can see why, and if we specify `total=False` on `_CheckInfoDict` it causes a bunch of other type errors in pebble.py. Yeah, I think what you've done here is reasonable -- just recording my thoughts.

Reply:
Thinking about this more, it does feel a bit like `_CheckInfoDict` should mark both `level` and `failures` as `NotRequired`, since if the dict passed to `from_dict` is missing those then that's handled. The type is being used to define the input to the method rather than being used to define what you get out (which is obviously `CheckInfo` itself).

Since we don't have `NotRequired` yet, it'd have to be done like this:

```python
_BaseCheckInfoDict(TypedDict):
    name: str
    status: Union['CheckStatus', str]
    threshold: int
_CheckInfoDict(_BaseCheckInfoDict, total=False):
    level: Optional[Union['CheckLevel', str]]
    failures: int
```

Would you be ok with me changing it to that (maybe minus the non-base class's underscore pending the other discussion), and then getting rid of a couple of the `type: ignore`s?

Reply:
Since we _do_ have `Required` (via `typing_extensions`) I'll go ahead with this change.